[doc][orc8r] Add documentation on AWS resources deployed for Orchestrator

joary opened the pull request:
<!--
    Tag your PR title with the components that it touches.
    E.g. "[lte][agw] Changeset" or "[orc8r][docker] ..."
-->

## Summary

<!-- Enumerate changes you made and why you made them -->
- Added documentation on Orc8r resources deployed on AWS
- Included documentation in docusaurus sidebar

## Test Plan

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->

![image](https://user-images.githubusercontent.com/1166157/117491170-39f71e00-af46-11eb-820c-1e905c00a116.png)

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

joary committed:
Add documentation on AWS resources deployed for Orchestrator

Signed-off-by: Joary Paulo Wanzeler Fortuna <joarypl@fb.com>

hcgatewood reviewed (changes requested):
Hey Joary thanks for adding this. Couple requests

High level
- In the test plan, include a screen cap of the page running locally. At minimum so we can see the included image
- Let's make a new "Architecture" subsection under "Orchestrator", i.e. `Orchestrator > Architecture`
- Then we can move your AWS info as well as this page https://magma.github.io/magma/docs/next/orc8r/intro into that section, which feels like a better home

Particular: some small nits for the documentation style
- Capitalization: make a pass to ensure consistency for the following capitalizations
    - K8s, Kubernetes
    - Orc8r, Orchestrator
    - NMS
    - Magma
    - Helm
- Don't use "here" as hyperlink text. The [chosen hyperlink text should be self-descriptive](https://www.nngroup.com/articles/writing-links/)

joary committed:
Create architecture subpage and correct capitalizations

Signed-off-by: Joary Paulo Wanzeler Fortuna <joarypl@fb.com>

joary committed:
Corrects minnor semantics

Signed-off-by: Joary Paulo Wanzeler Fortuna <joarypl@fb.com>

hcgatewood reviewed (changes requested):
Hey @joary, the Docusaurus layout was updated recently, so you'll need to rebase

Aside from that, content looks good. Consider taking a look at Magma Markdown conventions as part of your last-pass over the PR https://magma.github.io/magma/docs/next/contributing/contribute_conventions#markdown

Looking forward to merging this in!

hcgatewood commented:
Ping @joary

joary committed:
Move AWS Stack to Tech Reference

- Change documentation to reflect single DB
- Remove session on certificates and point to existing architecture doc
- Removed unexistent configuration fields

Signed-off-by: Joary Paulo Wanzeler Fortuna <joarypl@fb.com>

hcgatewood commented on docs/docusaurus/i18n/en.json:
```diff
@@ -283,6 +283,9 @@
         "title": "Experimental Orc8r Deployer"
       },
+      "orc8r/dev_aws_stack": {
+        "title": "Orchestrator AWS Stack"
```
```suggestion
        "title": "AWS Stack"
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+---
+id: dev_aws_stack
+title: Orchestrator AWS Stack
```
```suggestion
title: AWS Stack
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+hide_title: true
+---
+
+# Orchestrator Stack on AWS
```
```suggestion
# AWS Stack
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+---
+
+# Orchestrator Stack on AWS
+
```
```suggestion
This document describes the Orchestrator stack when it's deployed to AWS.

```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+
+![orc8r_aws_stack](assets/orc8r/orc8r_aws_stack.png)
+
+## What does terraform deploy in my AWS account?
```
```suggestion
## What does Terraform deploy in my AWS account?
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+
+## What does terraform deploy in my AWS account?
+
+The following resources are the basic infrastructure necessary to deploy Magma services:
```
```suggestion
The following resources are the basic infrastructure necessary to deploy Magma services
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+
+The following resources are the basic infrastructure necessary to deploy Magma services:
+
+* **Kubernetes cluster:** Orc8r runs over a Kubernetes (K8s) cluster, Magma uses Elastic Kubernetes Service (EKS) to deploy the cluster. Other resources are created to support this cluster.
```
```suggestion
* **Kubernetes cluster (EKS).** Orc8r runs over a Kubernetes (K8s) cluster, Magma uses Elastic Kubernetes Service to deploy the cluster. Other resources are created to support this cluster.
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+The following resources are the basic infrastructure necessary to deploy Magma services:
+
+* **Kubernetes cluster:** Orc8r runs over a Kubernetes (K8s) cluster, Magma uses Elastic Kubernetes Service (EKS) to deploy the cluster. Other resources are created to support this cluster.
+* **Elastic File System (EFS):** File system to hold persistent data for Kubernetes cluster.
```
Change all these colons to periods

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+|elasticsearch_version	|Optional	|ES version for ES domain.	|7.1	|
+
+The values for these configurations can be defined/overwritten in your main.tf `orc8r` module.
+Some configurations do not have default values and are mandatory to be present in your main.tf.
```
```suggestion
Some configurations do not have default values and are mandatory to be present in your `main.tf`.
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+|cluster_version	|Optional	|Kubernetes version for the EKS cluster.	|1.17	|
+|elasticsearch_version	|Optional	|ES version for ES domain.	|7.1	|
+
+The values for these configurations can be defined/overwritten in your main.tf `orc8r` module.
```
```suggestion
The values for these configurations can be defined/overwritten in your `main.tf` `orc8r` module.
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+|orc8r_tag	|Mandatory	|Image tag for Orchestrator components.	|-	|
+|deploy_openvpn	|Optional	|Flag to deploy OpenVPN server into cluster. This is useful if you want to remotely access AGWs.	|FALSE	|
+
+The values of these configurations can be defined/overwritten in your main.tf `orc8r-app` module.
```
```suggestion
The values of these configurations can be defined or overwritten in your `main.tf` file's `orc8r-app` module.
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+    1. FluentD: for log aggregation
+    2. Elastic Search Curator: to clean old logs
+2. **Metrics:** deploys all the services related to collect, store and show metrics, it includes:
+    1. Prometheus/Thanos for metric storage.
```
```suggestion
    1. Prometheus for metric storage.
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+5. **Secrets:** is used to apply a set of secrets required by the Magma Orchestrator.
+    1. [More details](https://github.com/magma/magma/blob/master/orc8r/cloud/helm/orc8r/charts/secrets/README.md)
+
+## Which certificates Magma use?
```
```suggestion
## Which certificates does Magma use?
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+
+## Which certificates Magma use?
+
+Orc8r needs certificates to assure messages traveling over the internet are encrypted and secure. Certificates are created in pairs with a public certificate (.crt or .pem) and a private key (.key). The former needs to be safely stored and kept secret, while the latter can be distributed to clients. Please [read the certificates section on Orchestrator architecture](https://magma.github.io/magma/ ...
```
```suggestion
Orc8r needs certificates to assure messages traveling over the internet are encrypted and secure. Certificates are created in pairs with a public certificate (`.crt` or `.pem`) and a private key (`.key`). The former needs to be safely stored and kept secret, while the latter can be distributed to clients. Please read the [certificates section on Orchestrator architecture](https://magma.github.io/magma/docs/next/orc8r/dev_security#certificates).
```

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,101 @@
+
+<span style="color:red">
+**Attention:** All the certificates are created with validity time period, so make sure you know when your certificates expire and schedule a maintenance to update them.
+</span>
```
```suggestion
> ***NOTE:*** All certificates are created with validity time period, so make sure you know when your certificates expire and schedule a maintenance to update them.
```

hcgatewood reviewed (approved):
Great thanks Joary. Accepted with a couple comments, please incorporate before merging 🙂

joary committed:
Small information and format changes

Signed-off-by: Joary Paulo Wanzeler Fortuna <joarypl@fb.com>

joary commented:
@hcgatewood I included all your comments, though I'm not authorized to merge.

hcgatewood commented on docs/readmes/orc8r/dev_aws_stack.md:
```diff
@@ -0,0 +1,100 @@
+
+The following resources are the basic infrastructure necessary to deploy Magma services
+
+* **Kubernetes cluster (EKS).* Orc8r runs over a Kubernetes (K8s) cluster, Magma uses Elastic Kubernetes Service to deploy the cluster. Other resources are created to support this cluster.
```
```suggestion
* **Kubernetes cluster (EKS).** Orc8r runs over a Kubernetes (K8s) cluster, Magma uses Elastic Kubernetes Service to deploy the cluster. Other resources are created to support this cluster.
```

hcgatewood reviewed (approved):
One last nit (formatting error) then I'll merge for you!

joary commented:
Thanks @hcgatewood, I solved it.

joary commented:
@hcgatewood , I had to rebase last two commits to include sign, DCO was failing for those.
